### PR TITLE
tests: add tests for util/ready package

### LINKED ISSUE #1271: util/ready/ready.go missing tests
I have observed strange behavior for this package in the production . Where deployment is ready and valid but it says it is not.
Would be good to test some edge cases.

### PR DESCRIPTION
**Pull request by troy0820**
### Which issue this PR addresses:

Fixes #1271 

### What this PR does / why we need it:
Adds tests to the package `pkg/util/ready`.  
Issue #1271 points out issues that this package experienced weird behavior.  With no unit tests, it was hard to pinpoint what was driving the weird behavior.  Adding tests will increase our coverage and create more confidence when using the functions supplied by this package.  

This PR will add tests to the package `pkg/util/ready`.  The package exports functions directly so no mocking is accomplished with interfaces. The mechanism used to tests these functions were using the built-in library that supports all the types referenced in the functions (i.e. corev1.Nodes, corev1.Pods, etc).  Also using `fake` clientsets provided by the k8s/openshift libraries allow the creation/failure of objects that are necessary to create errors for failing cases.  

### Test plan for issue:
`go tests -v ./pkg/util/ready`

### Is there any documentation that needs to be updated for this PR?
No documentation necessary as this will hopefully close issue #1271

### CODE REVIEW

File: pkg/util/ready/ready.go [hunk 1]
```diff
@@ -33,6 +33,8 @@ func NodeIsReady(node *corev1.Node) bool {
 func DaemonSetIsReady(ds *appsv1.DaemonSet) bool {
 	return ds.Status.DesiredNumberScheduled == ds.Status.NumberAvailable &&
 		ds.Status.DesiredNumberScheduled == ds.Status.UpdatedNumberScheduled &&
+		ds.Status.CurrentNumberScheduled == ds.Status.NumberReady &&
+		ds.Status.CurrentNumberScheduled > 0 &&
```
Comment (troy0820) [inline pkg/util/ready/ready.go:37]: This logic was added because if the daemonset was nil, it would say it's ready.

File: pkg/util/ready/ready_test.go [hunk 2]
```diff
@@ -0,0 +1,701 @@
...
+	ctx := context.Background()
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "running-pod",
+			Namespace: "default",
+		},
+	}
+
+	clientset := fake.NewSimpleClientset()
+	_, err := clientset.CoreV1().Pods("default").Create(ctx, pod, metav1.CreateOptions{})
```
Comment (troy0820) [inline pkg/util/ready/ready_test.go:478]: Creating pod covers case:
```go
 return PodIsRunning(p), nil
```

File: pkg/util/ready/ready_test.go [hunk 3]
```diff
@@ -0,0 +1,701 @@
...
+func TestCheckDaemonSetIsRunning(t *testing.T) {
+	ctx := context.Background()
+	ds := &appsv1.DaemonSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "running-ds",
+			Namespace: "default",
+		},
+	}
+	clientset := fake.NewSimpleClientset()
+	ds, err := clientset.AppsV1().DaemonSets("default").Create(ctx, ds, metav1.CreateOptions{})
```
Comment (troy0820) [inline pkg/util/ready/ready_test.go:426]: Creating covers case: 
```go
  return DaemonSetIsReady(ds), nil
```

File: pkg/util/ready/ready_test.go [hunk 4]
```diff
@@ -0,0 +1,701 @@
...
+func TestCheckMachineConfigPoolIsReady(t *testing.T) {
+	ctx := context.Background()
+
+	machineconfigpool := &mcv1.MachineConfigPool{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "machineconfigpool-not-found",
+		},
+	}
+	clientset := mcofake.NewSimpleClientset()
+	_, err := clientset.MachineconfigurationV1().MachineConfigPools().Create(ctx, machineconfigpool, metav1.CreateOptions{})
```
Comment (troy0820) [inline pkg/util/ready/ready_test.go:626]: Creating covers case: 
```go
return MachineConfigPoolIsReady(s), nil
```

File: pkg/util/ready/ready_test.go [hunk 5]
```diff
@@ -0,0 +1,701 @@
...
+	ctx := context.Background()
+
+	deployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "deployment-not-found",
+			Namespace: "default",
+		},
+	}
+	clientset := fake.NewSimpleClientset()
+	_, err := clientset.AppsV1().Deployments("default").Create(ctx, deployment, metav1.CreateOptions{})
```
Comment (troy0820) [inline pkg/util/ready/ready_test.go:528]: Creating cover case: 
```go
return DeploymentIsReady(d), nil
```

File: pkg/util/ready/ready_test.go [hunk 6]
```diff
@@ -0,0 +1,701 @@
...
+	ctx := context.Background()
+
+	statefulset := &appsv1.StatefulSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "statefulset-not-found",
+			Namespace: "default",
+		},
+	}
+	clientset := fake.NewSimpleClientset()
+	_, err := clientset.AppsV1().StatefulSets("default").Create(ctx, statefulset, metav1.CreateOptions{})
```
Comment (troy0820) [inline pkg/util/ready/ready_test.go:576]: Creating covers case:
```go
return StatefulSetIsReady(s), nil
```

**Review by mjudeikis — APPROVED** [review]
Good stuff :) Thanks!
